File: lectures/04-CPU-Machine-Code/upper.c
#include <stdio.h>

int main() {
  register int xr,acc;
  char mem[] = "Hello";

  for(xr=0;mem[xr];xr++) {
    acc = mem[xr];
    if ( (acc - 'a') < 0 ) continue;
    mem[xr] = acc - 0x20;
  }

  printf("%s\n", mem);
}

File: lectures/04-CPU-Machine-Code/upper.py

mem = "Hello"
newmem = ""

for xr in range(len(mem)):
  acc = mem[xr:xr+1]
  if acc >= 'a' and acc <= 'z' :
     acc = chr(ord(acc) - ord('a') + ord('A'))

  newmem = newmem + acc

print(newmem)

